src/main/java/org/breedinginsight/brapps/importer/services/processors/ExperimentProcessor.java

```diff
@@ -390,106 +475,175 @@ private String getObservationHash(String observationUnitName, String variableNam
+        Map<String, Trait> colVarMap = referencedTraits.stream().collect(Collectors.toMap(Trait::getObservationVariableName, Function.identity()));
+
+        Set<String> uniqueStudyAndObsUnit = new HashSet<>();
+        for (int rowNum = 0; rowNum < importRows.size(); rowNum++) {
+            ExperimentObservation importRow = (ExperimentObservation) importRows.get(rowNum);
+            PendingImport mappedImportRow = mappedBrAPIImport.get(rowNum);
+
+            if (StringUtils.isNotBlank(importRow.getGid())) { // if GID is blank, don't bother to check if it is valid.
+                validateGermplasm(importRow, validationErrors, rowNum, mappedImportRow.getGermplasm());
+            }
+
+            //Check if existing environment. If so, ObsUnitId must be assigned
+            if ((mappedImportRow.getStudy().getState() == ImportObjectState.EXISTING)
+                    && (StringUtils.isBlank(importRow.getObsUnitID()))) {
+                throw new MissingRequiredInfoException(MISSING_OBS_UNIT_ID_ERROR);
+            }
+
+            validateConditionallyRequired(validationErrors, rowNum, importRow);
+            validateObservationUnits(validationErrors, uniqueStudyAndObsUnit, rowNum, importRow);
+            validateObservations(validationErrors, rowNum, importRow, phenotypeCols, colVarMap, existingObsByObsHash);
         }
         return validationErrors;
     }
 
+    private void validateObservationUnits(ValidationErrors validationErrors, Set<String> uniqueStudyAndObsUnit, int rowNum, ExperimentObservation importRow) {
+        validateUniqueObsUnits(validationErrors, uniqueStudyAndObsUnit, rowNum, importRow);
+
+        String key = createObservationUnitKey(importRow);
+        PendingImportObject<BrAPIObservationUnit> ouPIO = observationUnitByNameNoScope.get(key);
+        if(ouPIO.getState() == ImportObjectState.NEW && StringUtils.isNotBlank(importRow.getObsUnitID())) {
+            addRowError(Columns.OBS_UNIT_ID, "Could not find observation unit by ObsUnitDBID", validationErrors, rowNum);
+        }
+    }
+
+    private Map<String, BrAPIObservation> fetchExistingObservations(List<Trait> referencedTraits, Program program) throws ApiException {
+        Set<String> ouDbIds = new HashSet<>();
+        Set<String> variableDbIds = new HashSet<>();
+        Map<String, String> variableNameByDbId = new HashMap<>();
+        Map<String, String> ouNameByDbId = new HashMap<>();
+        Map<String, String> studyNameByDbId = studyByNameNoScope.values()
+                                                                .stream()
+                                                                .map(PendingImportObject::getBrAPIObject)
+                                                                .collect(Collectors.toMap(BrAPIStudy::getStudyDbId, brAPIStudy -> Utilities.removeProgramKeyAndUnknownAdditionalData(brAPIStudy.getStudyName(), program.getKey())));
```

Comment:
When the experiment (and studies) being added are new, all the studies' StudyDbId are null. This leads to:

`2023-02-24 16:21:51.496-0500 [ForkJoinPool.commonPool-worker-5] ERROR o.b.b.i.services.FileImportService - Duplicate key
java.lang.IllegalStateException: Duplicate key
	at org.breedinginsight.brapps.importer.services.processors.ExperimentProcessor.fetchExistingObservations(ExperimentProcessor.java:525)
	at org.breedinginsight.brapps.importer.services.processors.ExperimentProcessor.validateFields(ExperimentProcessor.java:481)
	at org.breedinginsight.brapps.importer.services.processors.ExperimentProcessor.process(ExperimentProcessor.java:226)
	at org.breedinginsight.brapps.importer.services.processors.ProcessorManager.process(ProcessorManager.java:64)
	at org.breedinginsight.brapps.importer.model.imports.experimentObservation.ExperimentImportService.process(ExperimentImportService.java:72)
	at org.breedinginsight.brapps.importer.services.FileImportService.lambda$processFile$9(FileImportService.java:426)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run$$$capture(CompletableFuture.java:1771)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1763)
	at java.base/java.util.concurrent.ForkJoinTask.doExec$$$capture(ForkJoinTask.java:290)`

Reply:
Good catch!! Fixed and added a unit test for this